### PR TITLE
kubeadm: Bump minimum Kubernetes version to v1.12

### PR DESCRIPTION
**Pull request by rosti**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

Bump MinimumControlPlaneVersion and MinimumKubeletVersion to v1.12 and update
any related tests.
Part of the necessary release process.
Similar PRs for the previous three cycles: #68649 #63920 #57233

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
NONE

**Special notes for your reviewer**:
/cc @kubernetes/sig-cluster-lifecycle-pr-reviews
/area kubeadm
/assign @fabriziopandini 
/assign @timothysc 
/cc @neolit123 

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Review by neolit123 — APPROVED** [review]
@rosti thanks a lot.
and yeah we need to start using constants that define our skew policy.

File: cmd/kubeadm/test/cmd/init_test.go [hunk 1]
```diff
@@ -102,7 +102,7 @@ func TestCmdInitKubernetesVersion(t *testing.T) {
 		},
 		{
 			name:     "valid version is accepted",
-			args:     "--kubernetes-version=1.11.0",
+			args:     "--kubernetes-version=1.13.0",
```
Comment (idealhack) [inline cmd/kubeadm/test/cmd/init_test.go]: `1.12.0`?

Comment (rosti) [reply cmd/kubeadm/test/cmd/init_test.go]: Yep, too far bump for that, will fix it

**Comment by rosti** [comment]
The bazel test failure is due to latest k8s version being reported as 1.12 (when setup as `stable-1` in config). This should be automatically fixed upon release.

```
exec ${PAGER:-/usr/bin/less} "$0" || exit 1
Executing tests from //cmd/kubeadm/app/util/config:go_default_test
-----------------------------------------------------------------------------
--- FAIL: TestConfigFileAndDefaultsToInternalConfig (1.15s)
    --- FAIL: TestConfigFileAndDefaultsToInternalConfig/incompleteYAMLToDefaultedv1beta1 (0.03s)
        initconfiguration_test.go:123: the expected and actual output differs.
            	in: testdata/defaulting/master/incomplete.yaml
            	out: testdata/defaulting/master/defaulted.yaml
            	groupversion: kubeadm.k8s.io/v1beta1
            	diff: 
            --- expected
            +++ actual
            @@ -32,7 +32,7 @@
                 dataDir: /var/lib/etcd
             imageRepository: my-company.com
             kind: ClusterConfiguration
            -kubernetesVersion: v1.13.0
            +kubernetesVersion: v1.12.0
             networking:
               dnsDomain: cluster.global
               podSubnet: 10.148.0.0/16
            
FAIL
```

**Comment by neolit123** [comment]
```
-kubernetesVersion: v1.13.0
+kubernetesVersion: v1.12.0
```

https://github.com/kubernetes/kubernetes/commits/master/cmd/kubeadm/app/util/config/testdata/defaulting/master/incomplete.yaml

^ this is the history of this file.

i think, it should remain at v1.12.0 until we release and after that we should change - i.e. we should merge this PR.
the k version being dynamically fetched from the internet, while the version in the unit tests is hardcoded is not ideal.

**Comment by rosti** [comment]
Thanks @neolit123 for spotting this!
I was probably looking at something else, so I confused myself. Should be fixed now.

**Review by timothysc — APPROVED** [review]
/approve

**Comment by nikopen** [comment]
/lgtm


/kind bug 
needed for merging

**Review by luxas — APPROVED** [review]
/lgtm
/approve
Thanks @rosti!

**Pull request merged**